File: cpp/samchon/example/packer/ProductArray.cpp
#include <samchon/example/packer/ProductArray.hpp>

using namespace std;
using namespace samchon::library;
using namespace samchon::example::packer;

auto ProductArray::createChild(shared_ptr<XML>) -> Product*
{
	return new Product();
}

auto ProductArray::TAG() const -> string
{
	return "productArray";
}
auto ProductArray::CHILD_TAG() const -> string
{
	return "product";
}

File: ts/samchon/example/packer/ProductArray.ts
/// <reference path="../../protocol/EntityArray.ts" />

namespace samchon.example.packer
{
	export class ProductArray
		extends protocol.EntityArray<Product>
	{
		/* ------------------------------------------------------------------
			CONSTRUCTORS
		------------------------------------------------------------------ */
		/**
		 * Default Constructor.
		 */
		public constructor()
		{
			super();
		}

		protected createChild(xml: library.XML): Product
		{
			return new Product();
		}

		/* ------------------------------------------------------------------
			EXPORTERS
		------------------------------------------------------------------ */
		public TAG(): string
		{
			return "productArray";
		}
		public CHILD_TAG(): string
		{
			return "product";
		}
	}
}